app/packages/core/src/plugins/SchemaIO/components/ButtonView.tsx

```diff
@@ -33,25 +34,15 @@ function BaseButtonView(props) {
 }
 
 function OperatorButtonView(props) {
-  const { operator, params = {} } = props.schema.view;
-  const panel_state = useCustomPanelState();
-  const operatorExecutor = useOperatorExecutor(operator);
-  const promptForOperator = usePromptOperatorInput();
+  let { operator, params = {}, prompt } = props.schema.view;
+  const panelState = useCustomPanelState();
   const panelId = usePanelId();
+  const handleClick = usePanelEvent({ panelId, panelState });
   return (
     <BaseButtonView
       {...props}
       onClick={() => {
-        const actualParams = {
-          panel_id: panelId,
-          panel_state,
-          ...params,
-        };
-        if (props.schema.view.prompt) {
-          promptForOperator(operator, actualParams);
-        } else {
-          operatorExecutor.execute(actualParams);
```

Comment:
can we still keep this functionality as it may be used in operators. I would suggest creating a new component `PanelButtonView` and render it conditionally if `panelId` is available in component as Line 10